core/web/components/forms/profilePropertyRule/edit.tsx

```diff
@@ -281,6 +303,49 @@ export default function ({
                   <code>{opt.key}</code>: <small>{opt.description}</small>
                 </p>
 
+                {/* typeahead options */}
+                {opt.type === "typeahead" ? (
+                  <Typeahead
+                    id="typeahead"
+                    labelKey="key"
+                    onChange={(selected) => {
+                      if (selected.length === 1 && selected[0].key) {
+                        updateOption(opt.key, selected[0].key);
+                      }
+                    }}
+                    options={opt?.options}
+                    placeholder="Select a column"
+                    renderMenuItemChildren={(opt, props, idx) => {
+                      return [
+                        <span key={`opt-${idx}-key`}>
+                          {opt.key}
+                          <br />
+                        </span>,
+                        <small
+                          key={`opt-${idx}-examples`}
+                          className="text-small"
+                        >
+                          <em>
+                            Examples:{" "}
+                            {opt.examples
+                              ? opt.examples.slice(0, 3).join("").trim() !== ""
+                                ? opt.examples.slice(0, 3).join(", ")
+                                : "None"
+                              : null}
+                          </em>
+                        </small>,
+                      ];
+                    }}
+                    defaultSelected={
+                      profilePropertyRule.options[opt?.key]
```

Comment:
can be simplified as just `profilePropertyRule.options[opt?.key]` probably beccause that returns null or undefined itself.

Reply:
I tried that at first, but it freaked out.

Reply:
note that there's an array type or undefined (vs null).  This lib cares about the distinction!

Reply:
So maybe `profilePropertyRule.options[opt?.key] || undefined` then